lib/rps_v2.rb

```diff
@@ -0,0 +1,158 @@
+require_relative '../lib/player.rb'
+
+
+class RockPaperScissorsGame
+
+	#This method runs the game
+	def game_on
+		
+		puts "If you would like to play against the computer, please type 'yes'."
+		computer_game = gets.chomp.downcase
+
+		puts "Player one, please type your name."
+		@player1 = Player.new
+		@player1.set_p1_name(@player1.get_player_name)
+
+		if computer_game == "yes" 
+			@player2 = "Hal"
+
+		else
+			puts "Player two, please type your name."
+			@player2 = Player.new
+			@player2.set_p2_name(@player2.get_player_name)
+		end
+
+
+		puts "Please type how many rounds you would like to play."
+		number_of_games = gets.chomp.to_i
+
+		round_number = 0
+
+		#add this to player class .initialize?
+		score = {@player1 => 0, @player2 => 0}
+
+		while round_number < number_of_games
+			puts "#{@player1}" +", please type rock, paper, scissors, lizard, spock."
```

Comment:
`@player1` is the entire Player object–not just the name. The name is stored inside the Player object though, so you can write a method in the Player class that returns the name property.

Let's say you name that method `name_value`. You'd then be able to change this line to `puts "#{@player1.name_value}"`.